Unable to sort slice of `Vec<u8>`s

I expected the following code to just work:

```rust
extern crate afsort;

use afsort::AFSortable;

fn main() {
    let mut vecs: Vec<Vec<u8>> = vec![vec![1, 2], vec![1, 3]];
    afsort::sort_unstable_by(&mut vecs, |vec| vec.as_slice());
}
```

However, is doesn't compile because of `error[E0277]: the size for values of type `[u8]` cannot be known at compilation time`. This error arises from implicit `Sized` bound on `O` type of `sort_unstable_by` and `sort_req`. However, this bound is redundant since the code works only with values of type `&O`. I find this limitation pretty unfortunate since I can't fix on my side.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "afsort"
-version = "0.3.0"
+version = "0.3.1"
 authors = ["Anton Hägerstrand <anton.hagerstrand@gmail.com>"]
 exclude = [
     "test_resources/*"

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -320,7 +320,7 @@ fn ident<T>(t: &T) -> &T {
 #[inline]
 pub fn sort_unstable_by<T, O, S>(vec: &mut [T], sort_by: S)
 where
-    O: Ord + DigitAt,
+    O: Ord + DigitAt + ?Sized,
     S: Fn(&T) -> &O,
 {
     sort_req(vec, &sort_by, 0);
@@ -328,7 +328,7 @@ where
 
 fn sort_req<T, O, S>(vec: &mut [T], sort_by: &S, depth: usize)
 where
-    O: Ord + DigitAt,
+    O: Ord + DigitAt + ?Sized,
     S: Fn(&T) -> &O,
 {
     if vec.len() <= 32 {

